### PR TITLE
Fix Issues:

### PR DESCRIPTION
**Pull request by javeriaejaz-arbisoft**
- Add foreign key support in Registration Form
- Update migrations in applications and registration_extension

### CODE REVIEW

**Comment by zahid-arbisoft** [comment]
## Diff Coverage Report - ADG
Check-out complete **[report](https://app.circleci.com/pipelines/github/OmnipreneurshipAcademy/edx-platform/536/workflows/workflows_id/jobs/536/artifacts)**. Coverage !!! :rocket:
```
-------------
Diff Coverage
Diff: origin/develop...HEAD, staged and unstaged changes
-------------
openedx/adg/lms/applications/admin.py (100%)
openedx/adg/lms/applications/models.py (50.0%): Missing lines 54
openedx/adg/lms/registration_extension/forms.py (100%)
openedx/adg/lms/registration_extension/models.py (100%)
openedx/adg/lms/user_authn_override/login_form.py (100%)
-------------
Total:   11 lines
Missing: 1 line
Coverage: 90%
-------------
```

File: openedx/adg/lms/user_authn_override/login_form.py [hunk 1]
```diff
@@ -22,13 +23,13 @@ def get_login_session_form_override(request):
...
     3. Added placeholder for password
-
+    4. Added ModelChoiceField support to handle foreign key field
 
     Returns:
         FormDescription
 
     """
-
+    FormDescription.FIELD_TYPE_MAP[forms.ModelChoiceField] = "select"
```
Comment (zeshanarifios) [inline openedx/adg/lms/user_authn_override/login_form.py:32]: single quotes please

**Review by zeshanarifios — CHANGES_REQUESTED** [review]

**Review by zeshanarifios — APPROVED** [review]
LGTM 👍

**Comment by zahid-arbisoft** [comment]
## Diff Coverage Report - ADG
Check-out complete **[report](https://app.circleci.com/pipelines/github/OmnipreneurshipAcademy/edx-platform/538/workflows/workflows_id/jobs/538/artifacts)**. Coverage !!! :rocket:
```
-------------
Diff Coverage
Diff: origin/develop...HEAD, staged and unstaged changes
-------------
openedx/adg/lms/applications/__init__.py (100%)
openedx/adg/lms/applications/admin.py (100%)
openedx/adg/lms/applications/apps.py (100%)
openedx/adg/lms/applications/constants.py (100%)
openedx/adg/lms/applications/handlers.py (100%)
openedx/adg/lms/applications/helpers.py (100%)
openedx/adg/lms/applications/models.py (95.8%): Missing lines 54
openedx/adg/lms/applications/test/factories.py (100%)
openedx/adg/lms/applications/test/test_handlers.py (100%)
openedx/adg/lms/applications/test/test_helpers.py (100%)
openedx/adg/lms/registration_extension/forms.py (100%)
openedx/adg/lms/registration_extension/models.py (100%)
openedx/adg/lms/user_authn_override/login_form.py (100%)
-------------
Total:   103 lines
Missing: 1 line
Coverage: 99%
-------------
```

**Review by ahmed-faraz46 — APPROVED** [review]
LGTM 👍

**Pull request merged**